docs/components/search.vue

```diff
@@ -32,119 +31,64 @@
         />
       </svg>
     </button>
-    <b-popover target="bd-search-input" placement="bottom" triggers="focus">
-      <span v-if="search.length && Object.keys(results).length === 0">No results found</span>
-      <span v-else-if="search.length"></span>
-      <span v-else>Type something to start search</span>
-
-      <div
-        v-for="(results, section, idx) in results"
-        :key="section"
-        :class="idx > 0 ? 'mt-2' : ''"
-        role="group"
-      >
-        <h6 class="bd-text-purple my-1" v-html="section"></h6>
-        <div v-for="t in results" :key="t.href" class="my-1">
-          <b-link :to="t.href" @click="search = ''" v-html="t.title"></b-link>
-        </div>
-      </div>
-    </b-popover>
   </div>
 </template>
 
 <script>
-import groupBy from 'lodash/groupBy'
-import intersectionBy from 'lodash/intersectionBy'
-import { makeTOC } from '~/utils'
-import { components, directives, reference, misc } from '~/content'
-
-const SEARCH = []
-
-const process = (readme, section, page) => {
-  const baseURL = '/' + ['docs', section, page].filter(v => !!v).join('/')
-  const { title, toc } = makeTOC(readme)
-  ;[...toc].forEach(({ label, href }) => {
-    SEARCH.push({
-      section: title,
-      title: label,
-      href: `${baseURL}${href}`.replace('/#', '#')
-    })
-  })
-}
-
-// Async build the search database
-import('~/markdown/intro/README.md' /* webpackChunkName: "docs/intro" */).then(readme =>
-  process(readme.default, '', '')
-)
-Object.keys(components).forEach(page => {
-  import(`~/../src/components/${page}/README.md` /* webpackChunkName: "docs/components" */).then(
-    readme => process(readme.default, 'components', page)
-  )
-})
-Object.keys(directives).forEach(page => {
-  import(`~/../src/directives/${page}/README.md` /* webpackChunkName: "docs/directives" */).then(
-    readme => process(readme.default, 'directives', page)
-  )
-})
-Object.keys(reference).forEach(page => {
-  import(`~/markdown/reference/${page}/README.md` /* webpackChunkName: "docs/reference" */).then(
-    readme => process(readme.default, 'reference', page)
-  )
-})
-Object.keys(misc).forEach(page => {
-  import(`~/markdown/misc/${page}/README.md` /* webpackChunkName: "docs/misc" */).then(readme =>
-    process(readme.default, 'misc', page)
-  )
-})
+let scriptsInjected = false
 
 export default {
   data() {
     return {
-      search: ''
+      docsearch: null
     }
   },
-  computed: {
-    results() {
-      if (!this.search.length) {
-        return {}
+  mounted() {
+    this.loadDocsearch().then(this.initDocsearch)
```

Comment:
Since search doesn't appear on all docs pages (specifically `/` page).  navigating awy from a `/docs` pages to `/` will call `initDocsearch` again..  will that cause issues, or is it needed since the search input will be removed from the document and then put back in again.

Reply:
This component is only used and initialized when there is a search since it's the search.

Reply:
OK cool